openedx/core/djangoapps/programs/utils.py

```diff
@@ -340,13 +340,15 @@ def supplement_program_data(program_data, user):
             end_date = course_overview.end or datetime.datetime.max.replace(tzinfo=pytz.UTC)
             run_mode['start_date'] = start_date.strftime(human_friendly_format)
             run_mode['end_date'] = end_date.strftime(human_friendly_format)
+            run_mode['course_ended'] = end_date < timezone.now()
 
             run_mode['is_enrolled'] = CourseEnrollment.is_enrolled(user, course_key)
 
-            enrollment_start = course_overview.enrollment_start or datetime.datetime.min.replace(tzinfo=pytz.UTC)
+            enrollment_start = course_overview.enrollment_start or DEFAULT_START_DATE
```

Comment:
Here's the proof: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/access.py#L345.